### ISSUE TITLE
Pin the version of the Code of Conduct

### ISSUE THREAD

**Issue by jpscaletti** [atom/atom#8206]
Since its adoption, TODO Group has introduced incompatible changes to the code of conduct (explicitly allowing racism and sexism from members of groups they consider "marginalized". see http://todogroup.org/opencodeofconduct#definitions).

I propose to either fork the Coc or explicitly link to the version intended to use by the Atom project.

**Comment by beaugunderson** [comment]
I'm strongly in favor of using the latest version (currently https://github.com/todogroup/opencodeofconduct/commit/eb17e7745653861535099e7fef6fcb2c0402706e) because it very specifically [does *not* allow racism, sexism, etc.](https://github.com/todogroup/opencodeofconduct/blob/75bc62bb3578e8f6070946bdea2e14a9e766bf58/index.md#diversity-statement) while also protecting members of marginliazed groups from sanctions by members of dominant groups by refusing to act on complaints regarding "reverse -isms".

**Comment by cperciva** [comment]
@beaugunderson There's a semantic issue here: Some people rely on the historical dictionary definition of racism ("discrimination on the basis of race") while others have constructed a new definition based around power imbalances between groups (I'd be more precise, but I'm not aware of any precise formulation which has been widely accepted).  In the interest of constructive discourse, I'd suggest that it would be a good idea to be clearer about what we mean when we use particular words.

With regard to the code of conduct itself, I may be biased by my nationality here, but I think the approach taken in Canada is a good middle ground: Rather than declaring that some forms of discrimination aren't discrimination, say instead that some forms of discrimination -- "programs designed to ameliorate the conditions of disadvantaged individuals or groups", to borrow a phrase -- are acceptable.

**Comment by beaugunderson** [comment]
I just wanted to support using the latest version as it seems like it's getting better rather than worse. :)

I think your points re: defining racism are valid, though (I am among those who define racism around power imbalances or "power plus privilege").

**Comment by TFenby** [comment]
As a curious outsider: is this mostly an internal issue, or will you solicit feedback from users? See current feedback about using the newly-constructed definition of "-isms":
https://twitter.com/search?q=github%20code%20of%20conduct
https://twitter.com/github/status/623231837561819136

**Comment by jpscaletti** [comment]
Allow me to rephrase without using the redefined (and thus now useless) words

This new version of the CoC explicitly allows to some members of undefined "marginalized" groups to not to be bound to the code of conduct, and abuse and discriminate others solely based on their skin color and/or gender.

**Comment by beaugunderson** [comment]
@jpscaletti Can you explain how? A concrete example would be helpful, but I also don't think the CoC allows for abuse no matter who does it. What I think it does do is shield some people from a very specific type of complaint (one that can be used to harass).

**Comment by Balrogic** [comment]
I'm going to be perfectly frank, as I signed up this account for the sole purpose of sharing my point of view about your code of conduct. It is an abomination. I say that as someone that's pretty far on the political left, that has very strong feelings about the concepts of social justice and equality within society. Your code of conduct did not improve. It became substantially worse.

I feel discriminated against by your code of conduct for three reasons. One, because I'm white. Two, because I'm male. Three, because I'm autistic. I'm married to a black woman, by the way. My son is black. My grandson is black. Oh, and I'm LGBT too. In spite of being white and male, I was persecuted to extraordinary lengths due to being autistic. That persecution typically centered around my inability to socially integrate according to the expectations of others. I was physically assaulted dozens of times. I was punished whenever I attempted to seek help so the abuse would stop. I developed severe PTSD and agoraphobia as a direct result.

And you know what? I'd rather associate with the hostile sorts of anarchists you're trying to throw off the site than get caught up on here. Sure, they're assholes. The thing is, those assholes are assholes because they're assholes. It's not personal and I know that. I can tell them off whenever I please and they fully expect to be told off as necessary. Your rules, though? I don't expect to have a lot of success with social integration. I could contribute and help improve some projects, but given enough time working with the same people I'm going to start thinking of them as friends whether it's appropriate or not. The last thing I need is some power hungry madman coming at me with drama because I misconstrue a social signal or two along the way. Simple misunderstandings can be resolved with simple communication. Furthermore, I do *not* appreciate the way your rules would put a target on my back. I've lived enough of my life with a target on my back. I'm done with that.

I suggest you take great care in how you proceed, because you might find yourselves banning the wrong person, someone with a marginalized, federally protected disability which you justify through the lens of social convention which itself is highly oppressive toward that specific case.

**Comment by kevinsawicki** [comment]
See https://github.com/todogroup/opencodeofconduct/issues/84 for an update. Atom and Electron will be switching to a different code of conduct next week.

**Comment by CoWinkKeyDinkInc** [comment]
@kevinsawicki What would this be?  A fork or a independently written one?

**Comment by kevinsawicki** [comment]
> What would this be? A fork or a independently written one?

Not sure yet, going to look at a bunch of ones more closely and discuss more early next week.

**Comment by josteink** [comment]
May I suggest the simple "let's all act as adults" CoC?

Codifying anything more complex will 1. Lead to disagreements and 2. People gaming ("hacking") the rules and will lead to more troubles than in solves.

**Comment by CoWinkKeyDinkInc** [comment]
@josteink I agree with this.  All people really need as a CoC on Github of all places is to keep a calm and casual atmosphere.  It really hasn't been a problem on GitHub and I don't understand why they specifically implemented the OCoC when there is still many flaws in it.  People who would go offtopic with race, gender and whatever else will probably be blocked and reported by users and organizations anyway.

**Comment by f4w5yhq6z7bo** [comment]
May I recommend the No Code of Conduct (NCoC):

https://github.com/domgetter/NCoC

Read more here:

https://www.reddit.com/r/KotakuInAction/comments/3fr5d2/one_way_to_fight_back_against_entryist_sjw_codes/

**Comment by CoWinkKeyDinkInc** [comment]
@f4w5yhq6z7bo 
>We are all adults. Capable of having adult discussions.

Later down the list....
> CoC sounds like Cock, and I feel that this is a group of white males that is trying to downplay the seriousness of this issue in our community, and I boycott your movement, and am going to tell others to as well!

I'm not sure if the writers could be more pretentious or self-aware if what he is doing.

**Comment by josteink** [comment]
To be honest I think the only consequence of having a CoC in the first place is going to take something completely apolitical (code) and turn it into political discussions about which ism should have precedent over other isms (all in the name of good ofcourse)

It's generally going to be very unproductive with next to nothing to show for it as a result.

If this project needs to have a "CoC", then it needs to be as close to a null-coc as humanly possible, otherwise it will cause needless political fracturing which may cause harm to the project in other areas.

**Comment by f4w5yhq6z7bo** [comment]
> CoC sounds like Cock, and I feel that this is a group of white males that is trying to downplay the seriousness of this issue in our community, and I boycott your movement, and am going to tell others to as well!

That's part of the FAQ for the NCoC, not the NCoC itself.

For the actual NCoC itself, see:

https://github.com/domgetter/NCoC/blob/master/CODE_OF_CONDUCT.md

It's fairly short and to the point.

**Comment by CoWinkKeyDinkInc** [comment]
So?  They said that they are adults and willing to have "adult conversations".  Being condescending towards another is not "adult".  This worthless repository also assumes that all adults are mature, and only adults can be mature, when ironically whoever made this repo is not mature at all.

**Comment by f4w5yhq6z7bo** [comment]
The scope of the adult conversations is within the projects that the CoC will apply to. Some Frequently Asked Questions will obviously not be of an adult nature, but the project still has to respond to them, and they do so in the FAQ section.

I know you're smart enough to understand all of this, you're simply pretending to misunderstand for rhetorical leverage, and this is bad form. Please stop doing this.

**Comment by CoWinkKeyDinkInc** [comment]
So in other words, you don't know why?

**Comment by f4w5yhq6z7bo** [comment]
I have said no such thing. You are putting words in my mouth. Please stop arguing in bad faith.

**Comment by CoWinkKeyDinkInc** [comment]
Let me rephrase.  Do you know why NOCoC has a contradiction within the rules and the FAQ?

**Comment by krainboltgreene** [comment]
Please for the love of anything sacred just use a real code of conduct like http://contributor-covenant.org/

Using the TODO code of conduct or (ugh) NCOC is a slap in the face to a lot of developers who want safe and welcoming places.

**Comment by f4w5yhq6z7bo** [comment]
@CoWinkKeyDinkInc 

False premise. I cannot know something that is not true.

**Comment by benogle** [comment]
Ok ok. I'm going to lock this for now. We'll unlock this after thinking and discussing it.

**Comment by kevinsawicki** [comment]
There is now an open pull request to move from the Open Code of Conduct to the Contributor Covenant, see https://github.com/atom/atom/pull/8312 for more details.
